partiql-planner/src/main/kotlin/org/partiql/planner/internal/FnResolver.kt

```diff
@@ -147,7 +149,13 @@ internal object FnResolver {
     }
 
     private fun buildArgumentPermutations(args: List<StaticType>): List<List<StaticType>> {
-        val flattenedArgs = args.map { it.flatten().allTypes }
+        val flattenedArgs = args.map {
+            if (it is AnyOfType) {
+                it.flatten().allTypes.filter { it !is NullType }
+            } else {
+                it.flatten().allTypes
+            }
+        }
```

Comment:
What happens when the StaticType is AnyType? Would the following suffice?
```suggestion
        val flattenedArgs = args.map { it.flatten().allTypes }.filter { it !is NullType }
```

Reply:
Intuitively, if the input args is just a `StaticType.NULL`, then this will return an empty list, which might cause some trouble. 

I will look into this.